source/Mlos.Python/mlos/Spaces/HypergridAdapters/ContinuousToPolynomialBasisHypergridAdapter.py

```diff
@@ -0,0 +1,189 @@
+        # Record which adaptee dimensions are continuous
+        self._adaptee_contains_dimensions_to_transform = False
+        self._adaptee_dimension_names_to_transform = []
+        for adaptee_dimension in self._adaptee.dimensions:
+            if isinstance(adaptee_dimension, ContinuousDimension):
+                self._adaptee_dimension_names_to_transform.append(adaptee_dimension.name)
+        self._num_dimensions_to_transform = len(self._adaptee_dimension_names_to_transform)
+        self._adaptee_contains_dimensions_to_transform = self._num_dimensions_to_transform > 0
+
+        # see definition of _get_polynomial_feature_names() for usage
+        self._internal_feature_name_terminal_char = '_'
+
+        # Since sklearn PolynomialFeatures does not accept NaNs and these may appear in data frames from hierarchical hypergrids,
+        # the NaNs will be replaced with an imputed (finite) value.  The following sets the value used.
+        self._nan_imputed_finite_value = 0
+
+        # instantiate sklearn's polynomial features instance
+        self._polynomial_features = PolynomialFeatures(**self._polynomial_features_kwargs)
+        # because the exact number of additional dimensions that will be added depends on the parameters to sklearn's PF,
+        # *and* the sklearn PF instance above doesn't determine this information until after the .fit() method is called (requiring a dataframe),
+        # *and* the target hypergrid can not be constructed without knowing the resulting number of continuous dimensions,
+        # a trivial dataframe is constructed (all 1s) and .fit_transform() of _polynomial_features instance is called.
+        trivial_continuous_dim_x = np.ones((1, self._num_dimensions_to_transform))
+        trivial_polynomial_features_y = self._polynomial_features.fit_transform(trivial_continuous_dim_x)
+        self._polynomial_features_powers = self._polynomial_features.powers_
+        self._num_polynomial_basis_dimensions_in_target = trivial_polynomial_features_y.shape[1]
+        self._target_polynomial_feature_map = {}  # keys are target dimension names, values are index in features
+        self._build_simple_hypergrid_target()
+
+    def _build_simple_hypergrid_target(self) -> None:
+        self._target = SimpleHypergrid(
+            name=self._adaptee.name,
+            dimensions=None,
+            random_state=self._adaptee.random_state
+        )
+
+        # Add all adaptee dimensions to the target
+        # This aligns with this adapter's goal since the linear terms will always be included in the polynomial basis functions
+        for adaptee_dimension in self._adaptee.dimensions:
+            self._target.add_dimension(adaptee_dimension.copy())
+
+        if not self._adaptee_contains_dimensions_to_transform:
+            return
+
+        # add new dimensions to be created by sklearn PolynomialFeatures
+
+        # construct target dim names using adaptee dim names and polynomial feature powers matrix

```

Comment:
I'm not sure why this is necessary when passing the adaptee feature names to ``get_feature_names``.